terraform/modules/coreos_asg/main.tf

```diff
@@ -0,0 +1,223 @@
+data "aws_region" "selected" {
+    current = true
+}
+
+data "aws_caller_identity" "current" {}
+
+resource "aws_ecs_cluster" "cluster" {
+    name = "${var.name}"
+}
+
+data "aws_ami" "coreos_stable" {
```

Comment:
Also ContainerLinux is the officially supported base OS for k8s so I will probably opt to use that there too until your replacement exists.